openmmtools/forces.py

```diff
@@ -267,35 +283,67 @@ class RadiallySymmetricRestraintForce(utils.RestorableOpenMMObject):
+        assert len(restraint_parameters) == 1 or isinstance(restraint_parameters, collections.OrderedDict)
+        assert "Distance" not in restraint_parameters, ('"Distance" cannot be a restraint_parameter as it will ' 
+                                                        'clash with the CV Variable"')
+        assert "Distance" in energy_string, ('"Distance" must be in the energy_string as it will be the collective '
+                                             'variable the base CustomCVForce uses.')
+
+        # Augment energy_string with parameters
+        energy_string += ";"
+        for parameter_name, parameter_value in restraint_parameters.items():
+            # Ensure OpenMM Unit system
+            if isinstance(parameter_value, unit.Quantity):
+                parameter_value = parameter_value.in_unit_system(unit.md_unit_system)
+                restraint_parameters[parameter_name] = parameter_value
+                # Strip unit without overwriting dict entry efficiently
+                # Fetching value is faster than dividing by unit since
+                # we are already in correct unit system
+                parameter_value = parameter_value._value
+            energy_string += "{} = {};".format(parameter_name, parameter_value)
+        # Augment args with the energy string
+        args = (energy_string,) + args
+
+        # Construct
         super(RadiallySymmetricRestraintForce, self).__init__(*args, **kwargs)
         self._controlling_parameter_name = controlling_parameter_name
 
-        # Unzip bond parameters names and values from dict.
-        assert len(restraint_parameters) == 1 or isinstance(restraint_parameters, collections.OrderedDict)
-        parameter_names, parameter_values = zip(*restraint_parameters.items())
+        # Create the distance force used for the CV force wraps
+        distance_force = self._create_distance_force()
+        self.addCollectiveVariable("Distance", distance_force)
 
         # Let the subclass initialize its bond.
-        self._create_bond(parameter_values, restrained_atom_indices1, restrained_atom_indices2)
+        self._create_bond(restrained_atom_indices1, restrained_atom_indices2)
 
         # Add parameters.
         self.addGlobalParameter(self._controlling_parameter_name, 1.0)
-        for parameter in parameter_names:
-            self.addPerBondParameter(parameter)
+        self._restraint_parameters = restraint_parameters
 
     # -------------------------------------------------------------------------
     # Abstract methods.
     # -------------------------------------------------------------------------
 
     @abc.abstractmethod
-    def _create_bond(self, bond_parameter_values, restrained_atom_indices1,
+    def _create_distance_force(self):
+        """
+        Create and return the underlying force object
+
+        Returns
+        -------
+        force : OpenMM.Force
+            The force measuring distance to be used as the CV
+        """
+
```

Comment:
I didn't know that creating a function without at least `pass` was valid syntax!

Reply:
Neither did I. That was an oversight on my part. I think having a `pass` here is better anyways so its clear that this is an empty function and the end of the function.